src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -205,20 +247,54 @@ public void setChassisSpeeds(ChassisSpeeds chassisSpeeds) {
    * @param rot the angle to move to, in radians
    */
   public void runChassisPID(double x, double y, double rot) {
-    double xSpeed = m_xController.calculate(m_odometry.getPoseMeters().getX(), x);
-    double ySpeed = m_yController.calculate(m_odometry.getPoseMeters().getY(), y);
+    double xSpeed = m_xController.calculate(m_poseEstimator.getEstimatedPosition().getTranslation().getX(), x);
+    double ySpeed = m_yController.calculate(m_poseEstimator.getEstimatedPosition().getTranslation().getY(), y);
```

Comment:
you don't need to get the translaion, you can get the X and Y directly from the pose.